backend/dataall/base/utils/logs.py

```diff
@@ -0,0 +1,32 @@
+from typing import Optional, Callable, List, Any
+from dataall.base.config import config
+from dataall.base.utils.decorator_utls import process_func
+
+
+def is_feature_has_allowed_values(
+    allowed_values: List[Any],
+    default_value: Any,
+    resolve_property: Optional[Callable] = None,
+    config_property: Optional[str] = None,
+):
+    def decorator(f):
+        fn, fn_decorator = process_func(f)
+
+        def decorated(*args, **kwargs):
+            config_property_value = None
+            if config_property is None and resolve_property is None:
+                raise Exception('Config property not provided')
+            if resolve_property:
+                config_property_value = resolve_property(*args, **kwargs)
+            if config_property:
+                config_property_value = config_property
+            value = config.get_property(config_property_value, default_value)
+            if value not in allowed_values:
+                raise Exception(
+                    f'Disabled since incorrect values in config {config_property_value}. Correct config values {allowed_values}'
+                )
+            return fn(*args, **kwargs)
+
```

Comment:
I thought we had agreed to use this to enable/disable feature, not for only validation of config params. 

```
def is_feature_enabled_for_values(
    allowed_values: List[Any],
    enabled_values: List[Any],
    default_value: Any,
    resolve_property: Optional[Callable] = None,
    config_property: Optional[str] = None,
):
    def decorator(f):
        fn, fn_decorator = process_func(f)

        def decorated(*args, **kwargs):
            config_property_value = None
            if config_property is None and resolve_property is None:
                raise Exception('Config property not provided')
            if resolve_property:
                config_property_value = resolve_property(*args, **kwargs)
            if config_property:
                config_property_value = config_property
            value = config.get_property(config_property_value, default_value)
            if value not in allowed_values:
                raise Exception(
                    f'Disabled since incorrect values in config {config_property_value}. Correct config values {allowed_values}'
                )
            if value not in enabled_values:
                raise Exception(f'Disabled by config for {value=} . Enabled config values {enabled_values}')
            return fn(*args, **kwargs)

        return fn_decorator(decorated)

    return decorator
```
This removes the need to implement enabled/disabled logic in the particular shares/log/X-service

Reply:
@dlpzx, I was thinking that the decorator just be used as not a feature enable / disable checker but just to check if the values are correctly put. The logic to enable and disable would then be handled in the service layers. 
I like this idea in which it checks if the values are relating to enabled/disabled config values, will make the change. I forgot / didn't think like this